jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +308,37 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def cone_aperture(self):
+        # Retrieve flux cube and create an array to represent the cone mask
+        flux_cube = self._app._jdaviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D,
+                                                                          statistic=None)
+        # TODO: Replace with code for retrieving display_unit in cubeviz when it is available
+        display_unit = flux_cube.spectral_axis.unit
+        if display_unit.physical_type != 'length':
+            self.hub.broadcast(SnackbarMessage('Spectral axis unit physical type is '
+                                               f'{display_unit.physical_type}, must be length',
+                                               color="error", sender=self))
+            return
+
+        masks_float_values = np.zeros_like(flux_cube.flux.value, dtype=np.float32)
+
+        # Center is reverse coordinates
+        center = (self.aperture.selected_spatial_region.center.y,
+                  self.aperture.selected_spatial_region.center.x)
+
+        # Loop through cube and create cone aperture at each wavelength. Then convert that to a
+        # mask using the selected aperture method and add that to a mask cube.
+        # TODO: Use flux_cube.spectral_axis.to_value(display_unit) when we have unit conversion.
+        radii = ((flux_cube.spectral_axis.value / self.reference_wavelength) *
+                 self.aperture.selected_spatial_region.radius)
+        im_shape = (flux_cube.shape[1], flux_cube.shape[0])  # Reversed like center
```

Comment:
I think this is what you are trying to do, right, @javerbukh ? Hard to infer from the `len(,,,)` calls and the test cube was symmetrical so I cannot tell.